4a.allosigma_filtering/allosigma-filtering

```diff
@@ -285,8 +285,88 @@ def filter_on_distance_value_biopython(PDB, df, distance_threshold, file):
         exit(1)
     else:
         print(f"Filtered successfully on distance, removing sites closer than {distance_threshold} Å.")
-        return df   
+        return df
+
+def filter_on_sidechain_distance_value_biopython(PDB, df, distance_threshold):
+    """
+    A function that takes the dataframe produced by filter_on_dg(df, dg_threshold)
+    and filters out mutation–response site pairs where any side-chain heavy atoms
+    (or glycine sidechain hydrogen atoms) are closer than the defined distance_threshold. 
+
+    Parameters
+    ----------
+    df : A dataframe of mutations (x) and sites (y) with the allosigma
+    calculated dG score is larger than the dG threshold.
+    PDB : The PDB file name
+    distance_threshold: A float or integer that determines the minimum 
+    distance a long-range mutation is allowed to be.
+
+    Returns
+    -------
+    df : A dataframe where the local connection values are excluded.
+
+    """
+    pdb_id = op.splitext(PDB)[0]
+    structure = Bio.PDB.PDBParser().get_structure(pdb_id, PDB)[0]
+    #dictionary to store residue id and corresponding atom objects
+    residue_atoms = {}
+    #flag to check if all glycines have hydrogen atoms
+    all_gly_have_H = True
+    for residue in structure.get_residues():
+        if Bio.PDB.is_aa(residue):
+            residue_id = f"{seq1(residue.get_resname())}{residue.get_id()[1]}"
+            #list to store atom objects 
+            atoms = []
+            #handle glycine exception
+            if residue.get_resname() == "GLY":
+                #flag to check if each glycine has hydrogen atoms 
+                find_H = False
+                for atom in residue:
+                    if atom.get_name().startswith("HA"):
+                        atoms.append(atom)
+                        find_H = True
+                if not find_H:
+                    all_gly_have_H = False
+            #handle rest of the residues
+            else:
+                for atom in residue:
+                    if atom.element != "H" and atom.get_name() not in {"N", "CA", "C", "O"}:
+                        atoms.append(atom)
+            if atoms:
+                residue_atoms[residue_id] = atoms
```

Comment:
here you would want to error out if you haven;t found any atom; we cannot have a residue without side-chain